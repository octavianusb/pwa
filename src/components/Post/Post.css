.title {
  font-size: 3rem;
}

.title a {
  color: #2f89de;
}
.title a:hover, .title a:focus {
  color: #30699e;
  transition: color 0.4s ease;
}

.content {
  font-size: 1.8rem;
  line-height: 1.7;
}
