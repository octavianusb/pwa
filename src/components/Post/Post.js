import React from 'react'
import { Link } from 'react-router-dom'

import styles from './Post.css'

export default function Post (props) {
  return (
    <div>
      <h2 className={styles.title}>
        {
          props.titleLink
          ? <Link to={`/post/${props.slug}`}>{props.title}</Link> : props.title
        }
      </h2>
      <p className={styles.content}>{props.excerpt}</p>
    </div>
  )
}
