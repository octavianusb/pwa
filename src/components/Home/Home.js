import React from 'react'

// COMPONENTS
import Post from '../Post/Post'

export default function ({posts}) {
  return (
    <div>
      { posts.length
        ? posts.map(post => <Post {...post} key={post.slug} titleLink />)
        : <p>{'Awaiting posts'}</p>
      }
    </div>
  )
}
