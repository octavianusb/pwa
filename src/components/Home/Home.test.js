import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'
import Post from '../Post/Post'

describe('Home component', () => {
  it('should render posts as expected', () => {
    const post = [{
      "title": "lorem",
      "slug": "ipsum",
      "excerpt": "dolor",
      "content": "sit amet"
    },
    {
      "title": "sit amet",
      "slug": "dolor",
      "excerpt": "ipsum",
      "content": "lorem"
    }]
    const component = shallow(<Home posts={post} />)
    const tree = toJson(component)
    expect(component.find(Post).length).toBe(2)
    expect(tree).toMatchSnapshot()
  })

  it('shouldn\'t render posts', () => {
    const component = shallow(<Home posts={[]} />)
    const tree = toJson(component)
    expect(component.find(Post).length).toBe(0)
    expect(tree).toMatchSnapshot()
  })
})