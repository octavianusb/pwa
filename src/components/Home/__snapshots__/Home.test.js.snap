// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render posts as expected 1`] = `
<div>
  <Post
    content="sit amet"
    excerpt="dolor"
    slug="ipsum"
    title="lorem"
    titleLink={true}
  />
  <Post
    content="lorem"
    excerpt="ipsum"
    slug="dolor"
    title="sit amet"
    titleLink={true}
  />
</div>
`;

exports[`Home component shouldn't render posts 1`] = `
<div>
  <p>
    Awaiting posts
  </p>
</div>
`;
